lights-out.js

```diff
@@ -40,28 +39,60 @@ class LightsOut {
                 neighborArray.push([row, this.numCols - 1]);     
             }
         } else {
-            neighborArray.push([row, 0]);
-            neighborArray.push([row, col - 1]);
+            neighborArray.push([row, 0]), neighborArray.push([row, col - 1]);
         }
-
         return neighborArray;
     }
 
     registerClicks() {
-        for(let row = 0; row < this.numRows; row++) {
-            for(let col = 0; col < this.numCols; col++) {
+        for (let row = 0; row < this.numRows; row++) {
+            for (let col = 0; col < this.numCols; col++) {
                 const button = document.querySelector(`#${this.gameId} [data-row="${row}"][data-col="${col}"]`);
                 button.addEventListener('click', () => {
                     button.classList.toggle("on");
                     const neighbors = this.getNeighbors(row, col);
-                    
                     for (const neighbor of neighbors) {
                         const [r, c] = neighbor;
                         document.querySelector(`#${this.gameId} [data-row="${r}"][data-col="${c}"]`)
-                            .classList.toggle("on")
+                            .classList.toggle("on");
+                    }
+                    if (!button.textContent) {
+                        this.solve();
+                    } else {
+                        button.textContent = null;
                     }

```

Comment:
I'm pretty sure you can remove this conditional (if and else) and simply replace it with `this.solve();`.